File: game/modified_game_nim.cpp
//{ Driver Code Starts
// Initial Template for C++

#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
// User function Template for C++

class Solution {
public:
  int findWinner(int n, int A[]) {
    // code here
    unordered_map<int, int> m;
    for (int i = 0; i < n; ++i) {
      m[A[i]]++;
    }
    int total = 0;
    for (auto &p : m) {
      total += p.second % 2;
    }
    return total % 2 ? 2 : 1;
  }
};

//{ Driver Code Starts.

int main() {
  int t;
  cin >> t;
  while (t--) {
    int n;
    cin >> n;
    int A[n];
    for (int i = 0; i < n; i++)
      cin >> A[i];

    Solution ob;
    cout << ob.findWinner(n, A) << "\n";
  }
  return 0;
}
// } Driver Code Ends

File: bit/modified_game_nim.py
# User function Template for python3


class Solution:
    def findWinner(self, n, A):
        # code here
        now = 0
        for a in A:
            now ^= a

        if now == 0:
            return 1
        return 2 if n % 2 else 1


# {
# Driver Code Starts
# Initial Template for Python 3

if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        n = int(input())
        A = input().split()
        for itr in range(n):
            A[itr] = int(A[itr])

        ob = Solution()
        print(ob.findWinner(n, A))

# } Driver Code Ends
